third_party/ObjexxFCL/src/ObjexxFCL/vectorize.hh

```diff
@@ -23,16 +23,33 @@
 #define ASSUME(b)
 #define ASSUME_ALIGNED(p,b) p=(decltype(p))__builtin_assume_aligned(p,b)
 #define RESTRICT __restrict__
-#elif defined(_WIN32) && ( defined(_MSC_VER) || defined(__INTEL_COMPILER) )
+
+#elif defined(_WIN64) && ( defined(_MSC_VER) || defined(__INTEL_COMPILER) )
 #define ASSUME(b) __assume(b)
 #define ASSUME_ALIGNED(p,b) __assume_aligned(p,b)
 #define RESTRICT __restrict
+
+// Disable alignment hints for 32-bit windows builds
+#elif defined(_WIN32) && ( defined(_MSC_VER) || defined(__INTEL_COMPILER) )
+#define ASSUME(b) __assume(b)
+#define ASSUME_ALIGNED(p,b) 
+#define RESTRICT __restrict
+
 #else
 #define ASSUME(b)
 #define ASSUME_ALIGNED(p,b)
 #define RESTRICT
 #endif
 
+// Alignment for Compiler Options
+#if defined (__AVX2__)
```

Comment:
I believe @deadparrot put that in (I just copy-pasted from a different
file), but point well taken.

On Sat, Dec 12, 2015 at 10:07 PM, Mark Adams <notifications@github.com>
wrote:

> In third_party/ObjexxFCL/src/ObjexxFCL/vectorize.hh
> <https://github.com/NREL/EnergyPlus/pull/5278#discussion_r47439660>:
>
> >  #else
> >  #define ASSUME(b)
> >  #define ASSUME_ALIGNED(p,b)
> >  #define RESTRICT
> >  #endif
> >
> > +// Alignment for Compiler Options
> > +#if defined (__AVX2__)
>
> @amirroth <https://github.com/amirroth> AVX2 is not for __m512* types.
> AVX2 just adds integer operands and other features to AVX. It is AVX512
> that implements that new type. Also, unless someone is running E+ on Xeon
> Phi then no current processor implements AVX512 yet.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NREL/EnergyPlus/pull/5278/files#r47439660>.
>